chain/types/ethtypes/eth_transactions.go

```diff
@@ -36,189 +61,121 @@ type EthTx struct {
-		return EthTx{}, xerrors.Errorf("sender must be an eth account, was %s", smsg.Message.From)
+		return nil, fmt.Errorf("sender must be an eth account, was %s", smsg.Message.From)
 	}
 
-	// Probably redundant, but we might as well check.
+	// Ensure the signature type is delegated.
 	if smsg.Signature.Type != typescrypto.SigTypeDelegated {
-		return EthTx{}, xerrors.Errorf("signature is not delegated type, is type: %d", smsg.Signature.Type)
+		return nil, fmt.Errorf("signature is not delegated type, is type: %d", smsg.Signature.Type)
 	}
 
-	txArgs, err := EthTxArgsFromUnsignedEthMessage(&smsg.Message)
+	// Convert Filecoin address to Ethereum address.
+	_, err := EthAddressFromFilecoinAddress(smsg.Message.From)
 	if err != nil {
-		return EthTx{}, xerrors.Errorf("failed to convert the unsigned message: %w", err)
+		return nil, fmt.Errorf("sender was not an eth account")
 	}
 
-	r, s, v, err := RecoverSignature(smsg.Signature)
+	// Extract Ethereum parameters and recipient from the message.
+	params, to, err := getEthParamsAndRecipient(&smsg.Message)
 	if err != nil {
-		return EthTx{}, xerrors.Errorf("failed to recover signature: %w", err)
+		return nil, fmt.Errorf("failed to parse input params and recipient: %w", err)
 	}
 
-	from, err := EthAddressFromFilecoinAddress(smsg.Message.From)
-	if err != nil {
-		// This should be impossible as we've already asserted that we have an EthAddress
-		// sender...
-		return EthTx{}, xerrors.Errorf("sender was not an eth account")
-	}
-
-	return EthTx{
-		Nonce:                EthUint64(txArgs.Nonce),
-		ChainID:              EthUint64(txArgs.ChainID),
-		To:                   txArgs.To,
-		From:                 from,
-		Value:                EthBigInt(txArgs.Value),
-		Type:                 Eip1559TxType,
-		Gas:                  EthUint64(txArgs.GasLimit),
-		MaxFeePerGas:         EthBigInt(txArgs.MaxFeePerGas),
-		MaxPriorityFeePerGas: EthBigInt(txArgs.MaxPriorityFeePerGas),
-		AccessList:           []EthHash{},
-		V:                    v,
-		R:                    r,
-		S:                    s,
-		Input:                txArgs.Input,
-	}, nil
-}
-
-func EthTxArgsFromUnsignedEthMessage(msg *types.Message) (EthTxArgs, error) {
-	var (
-		to     *EthAddress
-		params []byte
-		err    error
-	)
-
-	if msg.Version != 0 {
-		return EthTxArgs{}, xerrors.Errorf("unsupported msg version: %d", msg.Version)
+	// Check for supported message version.
+	if smsg.Message.Version != 0 {
+		return nil, fmt.Errorf("unsupported msg version: %d", smsg.Message.Version)
 	}
 
-	if len(msg.Params) > 0 {
-		paramsReader := bytes.NewReader(msg.Params)
-		params, err = cbg.ReadByteArray(paramsReader, uint64(len(msg.Params)))
-		if err != nil {
-			return EthTxArgs{}, xerrors.Errorf("failed to read params byte array: %w", err)
+	// Process based on the signature data length.
+	switch len(smsg.Signature.Data) {
+	case 66: // Legacy Homestead transaction
+		if smsg.Signature.Data[0] != LegacyHomesteadEthTxSignaturePrefix {
```

Comment:
Is there a reason this first byte value has to be specified?  Is it just keeping things clean?  I don't see a real reason we can't just throw it away without checking.

Reply:
@ZenGround0 
1. We do plan to use this extended signature length for supporting more tx types down the road if need be. For now, there is only type with this extended length and that is the legacy transaction type.So, probably does not hurt to just explicity sanity check here. Either is fine.